src/java/org/apache/cassandra/dht/IPartitionerDependentSerializer.java

```diff
@@ -51,11 +41,8 @@ public interface IPartitionerDependentSerializer<T>
      */
     public T deserialize(DataInputPlus in, IPartitioner p, int version) throws IOException;
 
-    /**
-     * Calculate serialized size of object without actually serializing.
-     * @param t object to calculate serialized size
-     * @param version protocol version
-     * @return serialized size of object t
-     */
-    public long serializedSize(T t, int version);
+    default T deserialize(DataInputPlus in, int version) throws IOException
+    {
+        return deserialize(in, null, version);
```

Comment:
Would it make more sense to default to `IPartitioner.global`, rather than `null`?